### PR TITLE
fix(curriculum): dead links in data analyzer python

### LINKED ISSUE #47200: Dead link in demographic data analyzer
### Describe the Issue

There is dead link in [https://www.freecodecamp.org/news/learn-python-video-course/](https://www.freecodecamp.org/news/learn-python-video-course/) the file below
 
https://github.com/freeCodeCamp/freeCodeCamp/blob/1cd621fa673199b2fedf381fff2e9bd3a2797aa3/curriculum/challenges/english/08-data-analysis-with-python/data-analysis-with-python-projects/demographic-data-analyzer.md?plain=1#L16

Do we swap the link or delete it all together? or is it possible to get the link back?

Someone in Crowdin recommended https://www.freecodecamp.org/news/learn-python-basics-in-depth-video-course/ but that beginner video for project, I don't think it's fitting

### Affected Page

https://www.freecodecamp.org/learn/data-analysis-with-python/data-analysis-with-python-projects/demographic-data-analyzer

### PR DESCRIPTION
**Pull request by Sboonny**
Co-authored-by: mmatsumoto1026 <mmatsumoto1026@gmail.com>

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47200

### CODE REVIEW

**Review by moT01 — DISMISSED** [review]
I think I would be fine with this since we have a bunch of dead links out there. This link does seem to fit the data analysis section. I'm not sure if it's right for the two infosec courses - they were using the same old link, so why not the same new one I suppose.

Hey @mmatsumoto1026, does this link seem appropriate for the two infosec courses to you? I found this one https://www.freecodecamp.org/news/intermediate-python-course/ as well, but I'm not sure.

If you want @Sboonny, you could add the `10 hours` to all of these and make the spacing consistent (some have a space in front of the length, some not). It's not necessary, though.

**Comment by Sboonny** [comment]
@moT01 I do agree, but because of the possibility that someone skips the data analytic certification, I kept the beginner links

**Comment by mmatsumoto1026** [comment]
Hello, @moT01. Thanks for checking our PR.

To consider if the suggested video will fit the Information Security projects, I checked the contents of them. A brief summary is as follows FYI.


1. Stock Price Checker (JavaScript project)
MongoDB, handler/controller, security feature with js, test with js, data privacy.

1. Anonymous Message Board (JavaScript project)
MongoDB, handler/controller, security feature with js, test with js

1. Port Scanner (Python project)
Create a port scanner using Python, socket library, if-else, function.

1. SHA-1 Password Cracker (Python project)
Hashing using SHA-1 by importing hashlib library.

1. Secure Real Time Multiplayer Game (JavaScript project)
Using helmet@^3.21.3 for the user stories


In the Port Scanner and SHA-1 Password Cracker projects, It seems that a camper does not need to use Pandas, NumPy, Matplotlib, or Seaborn.

So while [Python for Everybody Video Course](https://www.youtube.com/watch?v=8DvywoWv6fI) will be able to teach basic knowledges for beginner to complete the two projects, I simply think that [Data Analysis with Python Course](https://www.youtube.com/watch?v=GPVsHOlRBBI&t=26401s) contains many topics which are not so relative for InfoSec projects although it also provide Python's basic knowledges.

Also, while [Intermediate Python Programming Course](https://www.youtube.com/watch?v=HGOBQPFzWKo&t=18097s) can provide basic knowledge of data structures, It provides more advanced ones and for beginners it may be too rich to think about the two projects.

It seems that how to build Port Scanner was showed in the Curriculum "Python for Penetration Testing course", and hashlib Python Standard Library document's link was provided in the SHA-1 PC project.

So how about consider putting only the link for [Python for Everybody Video Course](https://www.youtube.com/watch?v=8DvywoWv6fI) in the two InfoSec projects.

I apologize my long comments.

Thanks.

**Comment by moT01** [comment]
I reached out to Beau on this, he said the links I provided should be fine - but also that he thinks the original link that is now dead was this one: https://www.freecodecamp.org/news/learn-python-basics-in-depth-video-course/

We could use that on the infosec projects?

File: curriculum/challenges/english/09-information-security/information-security-projects/port-scanner.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ We are still developing the interactive instructional part of the Python curricu
 
 - <a href="https://www.freecodecamp.org/news/python-for-everybody/" target="_blank" rel="noopener noreferrer nofollow">Python for Everybody Video Course</a> (14 hours)
 
-- <a href="https://www.freecodecamp.org/news/learn-python-video-course/" target="_blank" rel="noopener noreferrer nofollow">Learn Python Video Course</a> (10 hours)
+- <a href="https://www.freecodecamp.org/news/learn-python-basics-in-depth-video-course/" target="_blank" rel="noopener noreferrer nofollow">Learn Python Basics in Depth</a> (4 hours)
+
+- <a href="https://www.freecodecamp.org/news/intermediate-python-course/" target="_blank" rel="noopener noreferrer nofollow">intermediate-python-course</a> (6 hours)
```
Comment (moT01) [inline curriculum/challenges/english/09-information-security/information-security-projects/port-scanner.md:20]:
```suggestion
- <a href="https://www.freecodecamp.org/news/intermediate-python-course/" target="_blank" rel="noopener noreferrer nofollow">Intermediate Python Course</a> (6 hours)
```

File: curriculum/challenges/english/09-information-security/information-security-projects/sha-1-password-cracker.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ We are still developing the interactive instructional part of the Python curricu
 
 - <a href="https://www.freecodecamp.org/news/python-for-everybody/" target="_blank" rel="noopener noreferrer nofollow">Python for Everybody Video Course</a> (14 hours)
 
-- <a href="https://www.freecodecamp.org/news/learn-python-video-course/" target="_blank" rel="noopener noreferrer nofollow">Learn Python Video Course</a> (10 hours)
+- <a href="https://www.freecodecamp.org/news/learn-python-basics-in-depth-video-course/" target="_blank" rel="noopener noreferrer nofollow">Learn Python Basics in Depth</a> (4 hours)
+
+- <a href="https://www.freecodecamp.org/news/intermediate-python-course/" target="_blank" rel="noopener noreferrer nofollow">intermediate-python-course</a> (6 hours)
```
Comment (moT01) [inline curriculum/challenges/english/09-information-security/information-security-projects/sha-1-password-cracker.md:20]:
```suggestion
- <a href="https://www.freecodecamp.org/news/intermediate-python-course/" target="_blank" rel="noopener noreferrer nofollow">Intermediate Python Course</a> (6 hours)
```

**Review by moT01 — APPROVED** [review]
LGTM @Sboonny 🎉

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**